contracts/test/CustomDecimalERC20Mock.sol

```diff
@@ -0,0 +1,21 @@
+// solium-disable linebreak-style
+pragma solidity 0.8.6;
+
+import "openzeppelin-contracts-sol8/token/ERC20/ERC20.sol";
+
+contract CustomDecimalERC20Mock is ERC20 {
+    uint8 private immutable _decimals;
+
+    constructor(
+        string memory _name,
+        string memory _symbol,
+        uint8 decimals_
+    ) ERC20(_name, _symbol) {
+        _mint(msg.sender, 200000000000000000000000000);
+        _decimals = decimals_;
```

Comment:
We add the decimal as a local state variable here. Would the behavior be different if we would use the openZeppelins method `_setupDecimals()` which they provide? Not sure if it would make a difference but if we would use real word tokens, they might have the decimals stored in a different way? I'm not so knowledgeable in this yet.
Link ([2](https://docs.openzeppelin.com/contracts/3.x/api/token/erc20#ERC20-_setupDecimals-uint8-)) to openZeppelin

Reply:
we are overriding the decimals function here. Didn't saw a method _setupDecimals() though. But as it's only for view, and the contracts are not affected by decimals. So I think, it won't cause any difference.

Reply:
openzeppelin have hardcoded 18 decimals.